ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java

```diff
@@ -399,10 +404,57 @@ public void clearRemovedTables(List<TableIdentifier> toRemoveTables) {
       LOG.info("OptimizeService init not completed, can't add new tables");
       return;
     }
-    toRemoveTables.forEach(this::clearTableCache);
+    toRemoveTables.forEach(tableIdentifier -> {
+      clearTableCache(tableIdentifier, true);
+      unOptimizeTables.remove(tableIdentifier);
+    });
     LOG.info("clear tables[{}] {}", toRemoveTables.size(), toRemoveTables);
   }
 
+  private int loadTablePoolSize() {
+    return Math.max(2, Runtime.getRuntime().availableProcessors());
+  }
+
+  private void enableSelfOptimizing(ArcticTable table) {
+    TableIdentifier tableIdentifier = table.id();
+    TableMetadata tableMetadata = buildTableMetadata(tableIdentifier);
+    if (!CompatiblePropertyUtil.propertyAsBoolean(
+        tableMetadata.getProperties(),
+        TableProperties.ENABLE_SELF_OPTIMIZING,
+        TableProperties.ENABLE_SELF_OPTIMIZING_DEFAULT)) {
+      if (!unOptimizeTables.contains(tableIdentifier)) {
+        unOptimizeTables.add(tableIdentifier);
+        persistTableRuntime(new TableOptimizeRuntime(tableIdentifier));
```

Comment:
Is it necessary to `persistTableRuntime` for an `unOptimizeTable`?

Reply:
> Is it necessary to `persistTableRuntime` for an `unOptimizeTable`?

I think we should be consistent since we are not deleting which tables are disabled for optimization. Also, we can maintain all tables currently available in catalog in the database.